test/utils/PhoneNumberValidationSpec.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.prop.Tables.Table
+import play.api.data.Form
+import play.api.data.Forms.single
+import tests.BaseUnitSpec
+import utils.PhoneNumberValidation.validatePhoneNumber
+
+class PhoneNumberValidationSpec extends BaseUnitSpec {
+
+  trait Setup {
+    val form = Form(single("phone" -> validatePhoneNumber))
+  }
+
+  "phone number validation" should {
+
+    "catch invalid length error" in new Setup {
+      val lengths = Table(
+        ("phone number", "validity"),
+        ("1", false),
+        ("12", false),
+        ("1234567890", false),
+        ("012345678901234567890", false), // 21
+        ("0123456789012345678901", false), // 22
+        ("01234567890", true), // 11
+        ("012345678901", true),
+        ("+44 1234567890", true),
+        ("+44 0808 157 0192", true),
+        ("+447904567897", true),
+        ("0790 305 1234", true),
+        ("0790-678-8992", true),
+        ("+44 790-590-5876", true)
+      )
+
+      TableDrivenPropertyChecks.forAll(lengths) { (phone, isValid) =>
+        val res: Form[String] = form.bind(Map("phone" -> phone))
+
+        if (isValid) {
+          res.hasErrors shouldBe false
+        } else {
+          res.errors(0).message shouldBe "error.phoneNumber.invalidLength"
+        }
+      }
+    }
+
+    "catch invalid format error" in new Setup {
+      val formats = Table(
+        ("phone number", "validity"),
+        ("phonenumber123456", false),
```

Comment:
I don't think in this case we need the Table. All entires are expected to be invalid. So we could just run a list of invalid values through the validator and assert that forAll is valid = false